api/v1alpha3/azuremachine_validation.go

```diff
@@ -50,7 +50,41 @@ func ValidateUserAssignedIdentity(identityType VMIdentity, userAssignedIdentetie
 	if identityType == VMIdentityUserAssigned && len(userAssignedIdenteties) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath, "must be specified for the 'UserAssigned' identity type"))
 	}
+	return allErrs
+}
+
+// ValidateDataDisks validates a list of data disks
+func ValidateDataDisks(dataDisks []DataDisk, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	lunSet := make(map[int32]struct{})
+	nameSet := make(map[string]struct{})
+	for _, disk := range dataDisks {
+		// validate that the disk size is between 4 and 32767.
+		if disk.DiskSizeGB < 4 || disk.DiskSizeGB > 32767 {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("DiskSizeGB"), "", "the Disk size should be a value between 1 and 2048"))
```

Comment:
"between 1 and 2048" seems like it should be "between 4 and 32767 GB", or the other way around.